docs/tutorials/text_sentiment_ngrams.ipynb

```diff
@@ -0,0 +1,419 @@
+    "\n",
+    "def valid_test_func(data_):\n",
+    "    _loss = 0\n",
+    "    _acc = 0\n",
+    "    data = DataLoader(data_, batch_size=BATCH_SIZE, collate_fn=generate_batch)\n",
+    "    for text, offsets, cls in data:\n",
+    "        text, offsets, cls = text.to(device), offsets.to(device), cls.to(device)\n",
+    "        with torch.no_grad():\n",
+    "            output = model(text, offsets)\n",
+    "            loss = criterion(output, cls)\n",
+    "            _loss += loss.item()\n",
+    "            _acc += (output.argmax(1) == cls).sum().item()\n",
+    "\n",
+    "    return _loss / len(data_), _acc / len(data_)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Split the dataset and run the model\n",
+    "Since the original AG_NEWS has no valid dataset, we have to split the training dataset into train/valid sets with the split ratios of 0.95 (train) and 0.05 (valid). Here we use `torch.utils.data.dataset.random_split` function in PyTorch core library.\n",
+    "\n",
+    "[CrossEntropyLoss](https://pytorch.org/docs/stable/nn.html?highlight=crossentropyloss#torch.nn.CrossEntropyLoss) criterion combines nn.LogSoftmax() and nn.NLLLoss() in a single class. It is useful when training a classification problem with C classes.\n",
+    "[SGD](https://pytorch.org/docs/stable/_modules/torch/optim/sgd.html) implements stochastic gradient descent method as optimizer. The initial learning rate is set to 4.0.\n",
+    "[StepLR](https://pytorch.org/docs/master/_modules/torch/optim/lr_scheduler.html#StepLR) is used here to adjust the learning rate through epochs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "import random\n",
+    "from torch.utils.data.dataset import random_split\n",
+    "N_EPOCHS = 5\n",
+    "min_valid_loss = float('inf')\n",
+    "\n",
+    "criterion = torch.nn.CrossEntropyLoss().to(device)\n",
+    "optimizer = torch.optim.SGD(model.parameters(), lr=4.0)\n",
+    "scheduler = torch.optim.lr_scheduler.StepLR(optimizer, 1, gamma=0.9)\n",
+    "\n",
+    "train_len = int(len(train_dataset) * 0.95)\n",
+    "sub_train_, sub_valid_ = \\\n",
+    "    random_split(train_dataset, [train_len, len(train_dataset) - train_len])\n",
+    "\n",
+    "for epoch in range(N_EPOCHS):\n",
+    "\n",
+    "    start_time = time.time()\n",
+    "    train_loss, train_acc = train_func(sub_train_)\n",
+    "    valid_loss, valid_acc = valid_test_func(sub_valid_)\n",
+    "\n",
+    "    _secs = int(time.time() - start_time)\n",
+    "    _mins = _secs / 60\n",
+    "    _secs = _secs % 60\n",
+    "\n",
+    "    print('Epoch: %d' %(epoch + 1), \" | time in %d minutes, %d seconds\" %(_mins, _secs))\n",
+    "    print(f'\\tLoss: {train_loss:.4f}(train)\\t|\\t{valid_loss:.4f}(valid)')\n",
+    "    print(f'\\tAcc: {train_acc * 100:.1f}%(train)\\t|\\t{valid_acc * 100:.1f}%(valid)')\n",
```

Comment:
This leads to misaligned text. It's a small detail, but could make the output much easier to read.